compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/window/ComposeWindow.web.kt

```diff
@@ -161,13 +168,16 @@ internal class ComposeWindow(
 
     private val canvasEvents = EventTargetListener(canvas)
 
+    private var keyboardModeState: KeyboardModeState = KeyboardModeState.Companion.Hardware
+
     private val platformContext: PlatformContext = object : PlatformContext {
         override val windowInfo get() = _windowInfo
 
         override val inputModeManager: InputModeManager = DefaultInputModeManager()
 
         override val textInputService = object : WebTextInputService() {
-            override fun resolveInputMode() = inputModeManager.inputMode
+            override fun isVirtualKeyboard() = keyboardModeState == KeyboardModeState.Companion.Virtual
```

Comment:
just wondering, is ".Companion." necessary here?
I remember seeing it only in java code that uses APIs wrritten in kotlin.

Reply:
@eymar for some mysterious reason yet to be investigated compilation failed for wasm without invoking Companion directly